Extensions/shuffle+.js

```diff
@@ -1,148 +1,269 @@
-        menuItem.isEnabled = isInjected;
-        injectFunctions(isInjected);
-    });
-
-    new Spicetify.Menu.SubMenu("Shuffle+", [autoShuffleMenu, playDiscographyMenu]).register();
-
-    function injectFunctions(bool) {
-        if (bool) {
-            Spicetify.Platform.PlayerAPI.play = (uri, origins, options) => {
-                if (options?.skipTo) {
-                    if (options.skipTo.index !== undefined) {
-                        playerPlayOGFunc(uri, origins, options);
-                        return;
-                    } else if (options.skipTo.pageIndex !== undefined) {
-                        uri.uri = options.skipTo.fallbackContextURI;
-                    } else {
-                        throw "No idea what to do. Please report on Github repo, specify which page you are in.";
-                    }
-                }
-                fetchAndPlay(uri.uri);
-            };
-        } else {
-            // Revert
-            Spicetify.Platform.PlayerAPI.play = playerPlayOGFunc;
-        }
+    await initShufflePlus();
+})();
+
+async function initShufflePlus() {
+    const { React } = Spicetify;
+    const { useState } = React;
+
+    async function getLocalStorageDataFromKey(key) {
+        return Spicetify.LocalStorage.get(key);
     }
 
-    // Text of notification when queue is shuffled successfully
-    /** @param {number} count */
-    const NOTIFICATION_TEXT = (count) => `Shuffled ${count} items!`;
+    async function setLocalStorageDataWithKey(key, value) {
+        Spicetify.LocalStorage.set(key, value);
+    }
 
-    const cntxMenu = new Spicetify.ContextMenu.Item(
-        "Play with Shuffle+",
-        (uris) => {
-            if (uris.length === 1) {
-                fetchAndPlay(uris[0]);
-                return;
+    async function getConfig() {
+        try {
+            const parsed = JSON.parse(await getLocalStorageDataFromKey("shufflePlus:settings"));
+            if (parsed && typeof parsed === "object") {
+                return parsed;
             }
+            throw "";
+        } catch {
+            await setLocalStorageDataWithKey("shufflePlus:settings", `{}`);
+            return { artistMode: "all", artistNameMust: false };
+        }
+    }
 
-            playList(shuffle(uris));
-        },
-        (uris) => {
-            if (uris.length === 1) {
-                const uriObj = Spicetify.URI.fromString(uris[0]);
-                switch (uriObj.type) {
-                    case Spicetify.URI.Type.SHOW:
-                    case Spicetify.URI.Type.PLAYLIST:
-                    case Spicetify.URI.Type.PLAYLIST_V2:
-                    case Spicetify.URI.Type.FOLDER:
-                    case Spicetify.URI.Type.ALBUM:
-                    case Spicetify.URI.Type.COLLECTION:
-                    case Spicetify.URI.Type.ARTIST:
-                        return true;
+    const CONFIG = await getConfig();
+    await saveConfig();
```

Comment:
Is it required to do this on startup?

Reply:
yes

Reply:
> yes

Why so? The config isn't changed until the menu item is appended much later on.

Reply:
> dont remove
> 
> ```js
> case Spicetify.URI.Type.ARTIST:
>                         return true;
> ```

I wasn't referring to it.

Reply:
> > yes
> 
> Why so? The config isn't changed until the menu item is appended much later on.

it is the init of the config like default settings will be stored. if you remove it the places with `CONFIG.something` wont work

Reply:
Understandable. Thank you.